### ISSUE TITLE
columnLayout: customGroupLayout resets on click

### ISSUE THREAD

**Issue by darrenfurr** [ag-grid/ag-grid#4240]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => see 'Providing a Reproducible Scenario'
[] feature request => do not use Github for feature requests, see 'Customers of ag-Grid'
[] support request => see 'Requesting Community Support'
```

**Current behavior**
When defining a customGroupLayout for columnLayout when you deselect or select a column the group layout resets back to the original layout.

**Reproduce**:
- Navigate here: https://www.ag-grid.com/javascript-grid-tool-panel-columns/#custom-column-layout
- Click Custom Group Layout in the example
- Then deselect a column

**Expected behavior**
The customGroupLayout should persist through clicks

**Please tell us about your environment:**

* **ag-Grid version:** X.X.X
24.1.0

* **Browser:**
Mozilla/5.0 (Macintosh; Intel Mac OS X 11_0_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.88 Safari/537.36

* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
React/es6

**Comment by makinggoodsoftware** [comment]
Hi,

Your ticket has been flagged as in_zendesk, it means that we recognize this as an issue worth investigating so we have moved it to our official support channel for customers (zendesk)

We will carry on there with the investigation and we will update you as soon as possible.

Thanks

**Comment by kiril-matev** [comment]
Hello!

Thank you for reporting this - much appreciated!

We already have this bug in our backlog and we are tracking it with the following reference and description:
**AG-4758 [Column Tool Panel] Calling api.setColumnDefs() resets the expanded state of column groups in column tool panel**

We have fixed this bug as part of our upcoming release - ag-grid v25, which is scheduled for release by December 18th.

I would recommend signing up for automatic release notifications for ag-grid. Then, following a release, check whether the item was implemented by looking for it in the changelog page.

Thanks again for bringing this up with us.

**Comment by makinggoodsoftware** [comment]
Hi,

In order to help us keep our issues queue clear, we are going to close this issue

The official way to check the status for an issue raised here is to check its AG-xxxx ID against our pipeline and changelog

…the pipeline (if not yet developed)

https://www.ag-grid.com/ag-grid-pipeline/

... or the changelog, (if the issue has been resolved)

https://www.ag-grid.com/ag-grid-changelog/

We keep issues raised in the pipeline through github open for a while so that the community can add some feedback to them, and close them when there is no activity for a while.

If you want to add further feedback to a closed ticket with this comment, we would recommend opening a new ticket as we don’t monitor closed issues.

**Comment by darrenfurr** [comment]
For anyone else who runs into this... 
I did confirm it is fixed in v25.0.0, BUT discovered that if you do not have this toolPanalParam set your customColumnLayout will be removed if someone drags to reorder a column:
`toolPanelParams: { suppressSyncLayoutWithGrid: true }`
